src/essential/container.jl

```diff
@@ -9,36 +15,18 @@ function TracedModel(
     model::Model,
     sampler::AbstractSampler,
     varinfo::AbstractVarInfo,
-    rng::Random.AbstractRNG
-) 
+    rng::Random.AbstractRNG,
+)
     context = SamplingContext(rng, sampler, DefaultContext())
-    evaluator = _get_evaluator(model, varinfo, context)
-    return TracedModel{AbstractSampler,AbstractVarInfo,Model,Tuple}(model, sampler, varinfo, evaluator)
+    args, kwargs = DynamicPPL.make_evaluate_args_and_kwargs(model, varinfo, context)
+    return TracedModel{AbstractSampler,AbstractVarInfo,Model,Tuple}(
+        model,
+        sampler,
+        varinfo,
+        (ArgsAndKwargsF(model.f), args, kwargs),
+    )
```

Comment:
After some digging, I thought the issue was a lack of `deepcopy` for the `model` when not explicitly present in `args`. So I tried making this into

```julia
(f::ArgsAndKwargsF)(model_arg::DynamicPPL.Model, args, kwargs) = f.f(model_arg, args...; kwargs...)
```
and
```julia
    return TracedModel{AbstractSampler,AbstractVarInfo,Model,Tuple}(
        model,
        sampler,
        varinfo,
        (ArgsAndKwargsF(model.f), args[1], args[2:end], kwargs),
    )
```
With that, the `deepcopy` is correctly triggered. But the error in the tests is still occurring :confused:

Reply:
AFAIK we don't. E.g. `TapedTask` uses kwargs to pass the `deepcopy_types`.

Reply:
Also, just for the record, if I do:
```julia
(model.f, args...)
```
i.e. drop the `kwargs` + the diversion completely, then everything works again.

Reply:
Moreover, it only seems to happen when combined with `Gibbs`. I haven't run into these issues when running PG on it's own.

Reply:
For reference, before I used to do:

```julia
struct ArgsAndKwargsF{F}
    f::F
end

(f::ArgsAndKwargsF)(args, kwargs) = f.f(args...; kwargs...)
```

and

```julia
    return TracedModel{AbstractSampler,AbstractVarInfo,Model,Tuple}(
        model,
        sampler,
        varinfo,
        (ArgsAndKwargsF(model.f), args, kwargs),
    )
```

in an attempt to also support `kwargs`, but this is causing strange behaviors when using something like `PG` in `Gibbs`. The above comments are referring to the code that was doing this.

Reply:
Also, now that we have `Core.kwcall`, it should be possible to support from Julia 1.9 on onwards in Libtask.jl